server/main_server/src/agent_main.cpp

```diff
@@ -0,0 +1,1326 @@
+                        break;
+                    }
+                }
+                else if (bytes_sent > 0) {
+                    bytes_to_send -= bytes_sent;
+                }
+            }
+
+            mySockClose(net_obj->socket_handle());
+            std::free(startupPack);
+            return 0;
+        }
+
+        if (startupPack->clientUser[0] != '\0') {
+            if (const auto ec = chkAllowedUser(startupPack->clientUser, startupPack->clientRodsZone); ec < 0) {
+                sendVersion(net_obj, ec, 0, nullptr, 0);
+                mySockClose(net_obj->socket_handle());
+                std::free(startupPack);
+                return 0;
+            }
+        }
+
+        //
+        // This is where we return to iRODS 4.3 logic.
+        //
+
+        RsComm rsComm{};
+
+        // Originally set via an environment variable in initRsCommWithStartupPack in iRODS 4.2+.
+        rsComm.sock = _socket_fd;
+
+        irods::experimental::log::set_error_object(&rsComm.rError);
+        irods::at_scope_exit release_error_stack{[] { irods::experimental::log::set_error_object(nullptr); }};
+
+        rsComm.thread_ctx = static_cast<thread_context*>(std::malloc(sizeof(thread_context)));
+
+        bool require_cs_neg = false;
+        int status = initRsCommWithStartupPack(&rsComm, startupPack, require_cs_neg); // This version just uses the startupPack.
+
+        // manufacture a network object for comms
+        ret = irods::network_factory(&rsComm, net_obj);
+        if (!ret.ok()) {
+            log_agent::error(PASS(ret).result());
+        }
+
+        if (status < 0) {
+            log_agent::error("initRsCommWithStartupPack error: [{}]", status);
+            sendVersion(net_obj, status, 0, nullptr, 0);
+            cleanupAndExit(status);
+        }
+
+        // TODO This is initialized during agent factory startup. Why do we create a new one after forking the agent?
+        //irods::re_plugin_globals.reset(new irods::global_re_plugin_mgr);
+        // TODO This line invokes the start operation of all rule engine plugins. If a plugin opens a resource
+        // that cannot be shared (e.g. a database connection), then we have to come up with another way of
+        // achieving our goal of no config changes taking effect until a config reload/restart.
+        //
+        // All we want to do is make it so that configuration is static on server startup. Can a pre-startup
+        // entry point help with that? It would be run by the agent factory and only setup state for later use.
+        // Things that are considered a limited resource would NOT be handled here. This would make reloading of
+        // plugin configuration easier because the pre-startup logic is independent and well-defined. However, that's
+        // not something we need to worry about since the reload logic for the server simply forks new processes.
+        irods::re_plugin_globals->global_re_mgr.call_start_operations();
```

Comment:
The TODO comment can be removed now that the REPs have setup/teardown functions.

Reply:
Done.